storage/src/mmr/mem.rs

```diff
@@ -1,22 +1,44 @@
 //! A bare-bones MMR structure without pruning and where all nodes are hashes & maintained in
 //! memory.
 //!
-//! TERMINOLOGY:
+//! # Terminology
 //!
 //! An MMR is a list of perfect binary trees of strictly decreasing height. The roots of these trees
-//! are called the "peaks" of the MMR.
+//! are called the "peaks" of the MMR. Each "element" stored in the MMR is represented by some leaf
+//! node in one of these perfect trees.
 //!
-//! Each "element" stored in the MMR is represented by some leaf node in one of these perfect trees.
+//! The nodes of the MMR are laid out within a single vector according to a post-order traversal of
+//! the MMR trees, starting from the from tallest tree to shortest. The "position" of a node in the
+//! MMR is defined as the 0-based index of the node in this vector. This implies the positions of
+//! elements, which are always leaves, may not be contiguous even if they were consecutively added.
 //!
-//! The "position" of a node in the MMR is defined as the 0-based index of the node in the
-//! underlying nodes vector. As the MMR is an append-only data structure, node positions never
-//! change and can be used as stable identifiers.
+//! As the MMR is an append-only data structure, node positions never change and can be used as
+//! stable identifiers.
 //!
 //! The "height" of a node is 0 for a leaf, 1 for the parent of 2 leaves, and so on.
 //!
 //! The "root hash" of an MMR is the result of hashing together the size of the MMR and the hashes
 //! of every peak in decreasing order of height.
-
+//!
+//! # Examples
+//!
+//! (Taken from  https://docs.grin.mw/wiki/chain-state/merkle-mountain-range/): After adding 11
+//! elements to an MMR, it will have 19 nodes total with 3 peaks corresponding to 3 perfect binary
+//! trees as pictured below, with nodes identified by their positions:
+//!
+//!```text
```

Comment:
nit: I think you should leave a space between `//!` and ````txt`

Reply:
done